Gems/EMotionFX/Code/Source/Editor/Plugins/Ragdoll/ColliderCapsuleManipulators.cpp

```diff
@@ -0,0 +1,209 @@
+#include <EMotionStudio/EMStudioSDK/Source/EMStudioManager.h>
+#include <Editor/ColliderContainerWidget.h>
+#include <Editor/Plugins/Ragdoll/ColliderCapsuleManipulators.h>
+
+namespace EMotionFX
+{
+    const Physics::CapsuleShapeConfiguration* GetCapsuleShapeConfiguration(const PhysicsSetupManipulatorData& physicsSetupManipulatorData)
+    {
+        if (!physicsSetupManipulatorData.HasCapsuleCollider())
+        {
+            return nullptr;
+        }
+
+        return azdynamic_cast<Physics::CapsuleShapeConfiguration*>(
+            physicsSetupManipulatorData.m_colliderNodeConfiguration->m_shapes[0].second.get());
+    }
+
+    Physics::CapsuleShapeConfiguration* GetCapsuleShapeConfiguration(PhysicsSetupManipulatorData& physicsSetupManipulatorData)
+    {
+        return const_cast<Physics::CapsuleShapeConfiguration*>(
+            GetCapsuleShapeConfiguration(const_cast<const PhysicsSetupManipulatorData&>(physicsSetupManipulatorData)));
+    }
+
+    ColliderCapsuleManipulators::ColliderCapsuleManipulators()
+    {
+        m_adjustColliderCallback = AZStd::make_unique<PhysicsSetupManipulatorCommandCallback>(this, false);
+        EMStudio::GetCommandManager()->RegisterCommandCallback("AdjustCollider", m_adjustColliderCallback.get());
+    }
+
+    ColliderCapsuleManipulators::~ColliderCapsuleManipulators()
+    {
+        EMStudio::GetCommandManager()->RemoveCommandCallback(m_adjustColliderCallback.get(), false);
+    }
+
+    void ColliderCapsuleManipulators::Setup(PhysicsSetupManipulatorData& physicsSetupManipulatorData)
+    {
+        m_physicsSetupManipulatorData = physicsSetupManipulatorData;
+
+        if (!m_physicsSetupManipulatorData.HasCapsuleCollider())
+        {
+            return;
+        }
+
+        SetupCapsuleManipulators(EMStudio::g_animManipulatorManagerId);
+
+        AZ::TickBus::Handler::BusConnect();
+        PhysicsSetupManipulatorRequestBus::Handler::BusConnect();
+    }
+
+    void ColliderCapsuleManipulators::Refresh()
+    {
+        if (m_physicsSetupManipulatorData.HasCapsuleCollider())
+        {
+            UpdateCapsuleManipulators();
+        }
+    }
+
+    void ColliderCapsuleManipulators::Teardown()
+    {
+        if (!m_physicsSetupManipulatorData.HasCapsuleCollider())
+        {
+            return;
+        }
+
+        PhysicsSetupManipulatorRequestBus::Handler::BusDisconnect();
+        AZ::TickBus::Handler::BusDisconnect();
+        TeardownCapsuleManipulators();
+    }
+
+    void ColliderCapsuleManipulators::ResetValues()
+    {
+        if (m_physicsSetupManipulatorData.HasCapsuleCollider())
+        {
+            BeginEditing();
+            ResetCapsuleManipulators();
+            FinishEditing();
+            Refresh();
+        }
+    }
+
+    void ColliderCapsuleManipulators::BeginEditing()
+    {
+        if (!m_commandGroup.IsEmpty())
+        {
+            return;
+        }
+        m_commandGroup.SetGroupName("Adjust collider");
```

Comment:
I can't fully remember if we can actually reuse command groups without clearing them upfront. Can you please check if the command group here is empty after having used it before?

Reply:
Is that not what's happening in line 94? I just copied this pattern from another place commands were being used.